third_party/move/move-core/types/src/account_address.rs

```diff
@@ -276,19 +280,29 @@ impl TryFrom<String> for AccountAddress {
     type Error = AccountAddressParseError;
 
     fn try_from(s: String) -> Result<AccountAddress, AccountAddressParseError> {
-        Self::from_hex(s)
+        Self::from_str(&s)
     }
 }
 
 impl FromStr for AccountAddress {
     type Err = AccountAddressParseError;
 
+    /// Parse a string into an AccountAddress. Conforms to AIP-40, meaning addresses in
+    /// both SHORT and LONG form, with or without the 0x prefix, are accepted. It opts
+    /// for a permissive approach, where even non-special (as defined above) SHORT
+    /// string representations of addresses are accepted (to maximize compatibility).
+    /// This is allowed by AIP-40 at the discretion of the implementation.
     fn from_str(s: &str) -> Result<Self, AccountAddressParseError> {
-        // Accept 0xADDRESS or ADDRESS
-        if let Ok(address) = AccountAddress::from_hex_literal(s) {
-            Ok(address)
+        if !s.starts_with("0x") {
+            if s.is_empty() {
+                return Err(AccountAddressParseError);
+            }
```

Comment:
Similarly for this I think?

Reply:
It doesn't actually (if I remember correctly, from when I wrote the tests), and I don't want to change `from_hex_literal`.